Nodejs/Product/Analysis/Analysis/Analyzer/AnalysisUnit.cs

```diff
@@ -188,9 +188,9 @@ internal AnalysisUnit(JsAst ast, EnvironmentRecord environment)
         }
 
         internal virtual void AnalyzeWorker(DDG ddg, CancellationToken cancel) {
-            if (Tree != ProjectEntry.Tree) {
-                // we were enqueued and a new version became available, don't re-analyze against
-                // the old version.
+            if (Ast != null && ProjectEntry != null && ProjectEntry.Tree != null && Tree != ProjectEntry.Tree) {
```

Comment:
I couldn't reproduce this issue, so I'm not entirely sure (there are null checks for these properties elsewhere in the code, so I presumed it would be safe enough here.) This is something I plan on asking dino today before merging in.